### PR TITLE
Improve types

### PR DESCRIPTION
**Pull request by guilhermebodin**

### CODE REVIEW

File: src/diagnostics.jl [hunk 1]
```diff
@@ -48,13 +48,13 @@ function residuals(ss::StateSpace)
...
 """
-    jarquebera(e::Matrix{Float64})
+    jarquebera(e::Matrix{T})
 
 Run Jarque-Bera normality test and return p-values
 """
-function jarquebera(e::Matrix{Float64})
+function jarquebera(e::Matrix{T}) where T <: AbstractFloat
     n, p = size(e)
-    pvalue = Vector{Float64}(undef, p)
```
Comment (raphaelsaavedra) [inline src/diagnostics.jl]: I think p-value should always be `Float64` for simplicity

File: src/diagnostics.jl [hunk 2]
```diff
@@ -70,14 +70,14 @@ function jarquebera(e::Matrix{Float64})
...
-    ljungbox(e::Matrix{Float64}; maxlag = 20)
+    ljungbox(e::Matrix{T}; maxlag = 20)
 
 Run Ljung-Box independence test and return p-values
 """
-function ljungbox(e::Matrix{Float64}; maxlag = 20)
+function ljungbox(e::Matrix{T}; maxlag::Int = 20) where T <: AbstractFloat
     n, p = size(e)
     n <= maxlag && (maxlag = n-1)
-    pvalue = Vector{Float64}(undef, p)
```
Comment (raphaelsaavedra) [inline src/diagnostics.jl]: Here too

File: src/diagnostics.jl [hunk 3]
```diff
@@ -89,16 +89,16 @@ function ljungbox(e::Matrix{Float64}; maxlag = 20)
...
 
 Run homoscedasticity test and return p-values
 """
-function homoscedast(e::Matrix{Float64})
+function homoscedast(e::Matrix{T}) where T <: AbstractFloat
     n, p = size(e)
     t1 = floor(Int, n/3)
     t2 = ceil(Int, 2*(n/3)+1)
     dist = FDist(n-t2+1, t1)
-    pvalue = Vector{Float64}(undef, p)
```
Comment (raphaelsaavedra) [inline src/diagnostics.jl]: Here too

File: src/kalman.jl [hunk 4]
```diff
@@ -151,18 +151,18 @@ function statespace_covariance(psi::Vector{T}, p::Int, r::Int,
...
-    sqrtQ = kron(Matrix{Float64}(I, Int(r/p), Int(r/p)), tril!(ones(p, p)))
+    sqrtQ = kron(Matrix{T}(I, Int(r/p), Int(r/p)), tril!(ones(p, p)))
     sqrtQ[findall(isequal(1), sqrtQ)] = psi[(unknownsH+1):Int(unknownsH + (r/p)*(p*(p + 1)/2))]
 
     # Obtain full matrices
     H = gram(sqrtH)
     Q = gram(sqrtQ)
 
-    return H, Q
+    return T.(H), T.(Q)
```
Comment (raphaelsaavedra) [inline src/kalman.jl]: This is probably affecting performance as we're doing a cast in every estimation step, perhaps there's a better way?

**Comment by raphaelsaavedra** [comment]
We might want to benchmark this PR to see how much these changes have affected performance

File: src/kalman.jl (hunk 4 above)
Comment (guilhermebodin) [reply src/kalman.jl]: hmm true, I think it does not make much difference but let us check it

**Comment by guilhermebodin** [comment]
Some quick benchmarking showed me that the code has the same performace. We could study how MOI did for benchmarking performance in their testset

**Pull request merged**